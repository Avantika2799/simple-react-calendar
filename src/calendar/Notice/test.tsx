import { cleanup, render, RenderResult } from '@testing-library/react'
import React from 'react'

import Notice, { Props } from './Notice'

afterEach(cleanup)

describe('Notice', () => {
  let api: RenderResult, props: Props

  beforeEach(() => {
    props = {
      blockClassName: 'test-class-name',
      type: 'overlapping_with_disabled',
    }
  })

  describe('when type is `overlapping_with_disabled`', () => {
    beforeEach(() => {
      api = render(<Notice {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })

  describe('when type is `disabled_day_click`', () => {
    beforeEach(() => {
      props = {
        ...props,
        type: 'disabled_day_click',
      }
      api = render(<Notice {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })

  describe('when type is `null`', () => {
    beforeEach(() => {
      props = {
        ...props,
        type: null,
      }
      api = render(<Notice {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })
})
