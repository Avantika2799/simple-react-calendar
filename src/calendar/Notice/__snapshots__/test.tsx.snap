// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notice when day is \`Sunday\` default render 1`] = `
<div>
  <div
    class="test-class-name-notice"
  />
</div>
`;

exports[`Notice when type is \`disabled_day_click\` default render 1`] = `
<div>
  <div
    class="test-class-name-notice"
  >
    Selected range must not overlap with disabled dates.
  </div>
</div>
`;

exports[`Notice when type is \`null\` default render 1`] = `
<div>
  <div
    class="test-class-name-notice"
  />
</div>
`;

exports[`Notice when type is \`overlapping_with_disabled\` default render 1`] = `
<div>
  <div
    class="test-class-name-notice"
  />
</div>
`;
